factor/grace/classes/factor.php

```diff
@@ -135,7 +135,9 @@ public function get_state($redirectable = true) {
                             }
 
                             // We should never redirect if we have already passed.
-                            if ($redirectable && \tool_mfa\manager::get_cumulative_weight() >= 100) {
+                            // Do not stop on unknown, to avoid factors e.g. login banner from
+                            // hiding passing factors lower down in the list.
+                            if ($redirectable && \tool_mfa\manager::get_cumulative_weight(false) >= 100) {
```

Comment:
There is a get_total_weight method in the manager, and I think it duplicates the behaviour of get_cumulative with the stop on unknown flag = false, so I think this one can be simple. remove the stoponunknown flag, and call get_total_weight instead.

Reply:
Yep good catch - done